api/desecapi/tests/testdomains.py

```diff
@@ -177,13 +196,48 @@ def testDomainDetailURL(self):
 
 class AuthenticatedDynDomainTests(APITestCase):
     def setUp(self):
+        httpretty.reset()
+        httpretty.disable()
         if not hasattr(self, 'owner'):
             self.owner = utils.createUser(dyn=True)
             self.ownedDomains = [utils.createDomain(self.owner, dyn=True), utils.createDomain(self.owner, dyn=True)]
             self.otherDomains = [utils.createDomain(), utils.createDomain()]
             self.token = utils.createToken(user=self.owner)
             self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
 
+    def testCanDeleteOwnedDynDomain(self):
+        httpretty.enable()
+        httpretty.register_uri(httpretty.DELETE, settings.NSLORD_PDNS_API + '/zones/' + self.ownedDomains[1].name + '.')
+        httpretty.register_uri(httpretty.DELETE, settings.NSMASTER_PDNS_API + '/zones/' + self.ownedDomains[1].name+ '.')
+        httpretty.register_uri(httpretty.PATCH, settings.NSLORD_PDNS_API + '/zones/dedyn.io.')
+
+        url = reverse('domain-detail', args=(self.ownedDomains[1].pk,))
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
`DELETE` requests should be idempotent, that requires us to return 200/204 if the resource requested does not exist: https://tools.ietf.org/html/rfc7231#page-23

I can see though that some interpretations of this RFC may conclude that the idempotency does not extend to the response code given. However, this would only require the client to handle more reponse codes with the same meaning; hence I'd return 200/204 for deleted and non-existing resources, 403 for domains of other users. (Unless there is a good point why this is an unacceptable privacy intrusion.)